src/GitServer/pack-objects/BufferCursor.ts

```diff
@@ -0,0 +1,64 @@
+// Modeled after https://github.com/tjfontaine/node-buffercursor
+// but with the goal of being much lighter weight.
+export class BufferCursor {
```

Comment:
Is this really needed? Seems like a simpler abstraction is possible.

Reply:
Also as a general style, define your class by itself, and run `export default...` at the end.

Reply:
> Is this really needed? Seems like a simpler abstraction is possible.

Yes this should be possible with the [feross/buffer](https://www.npmjs.com/package/buffer)

> Also as a general style, define your class by itself, and run `export default...` at the end.

Oh yes, I forgot this for the imported code. This has been done for all classes :+1:

Reply:
Upon further inspection, this `BufferCursor` class provides some really helpful methods and I think if we remove it, we would make the `GitPackIndex` class much more complicated. So I will leave it in.